File: shared/src/memory.ts
export type MemoryEntry = {
  id: string;
  content: string;
  embedding: number[];
  source: string;
  timestamp: number;
  tags: string[];
};

export type SearchResult = {
  entry: MemoryEntry;
  score: number;
  matchType: "vector" | "keyword" | "hybrid";
};

export type DomainPattern = {
  id: string;
  pattern: string;
  confidence: number;
  source: string;
  createdAt: number;
  updatedAt: number;
};

export type DomainCorrelation = {
  id: string;
  sourceA: string;
  sourceB: string;
  rule: string;
  confidence: number;
  createdAt: number;
};

export type DomainThreshold = {
  id: string;
  source: string;
  metric: string;
  value: number;
  direction: "above" | "below";
  updatedAt: number;
};

export type MemoryEvent = {
  type: "created" | "updated" | "deleted";
  entryId: string;
  timestamp: number;
};

File: src-tauri/src/types/memory.rs
use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MemoryEntry {
    pub id: String,
    pub content: String,
    pub embedding: Vec<f32>,
    pub source: String,
    pub timestamp: u64,
    pub tags: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct SearchResult {
    pub entry: MemoryEntry,
    pub score: f64,
    pub match_type: MatchType,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum MatchType {
    Vector,
    Keyword,
    Hybrid,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct MemoryEvent {
    #[serde(rename = "type")]
    pub event_type: MemoryEventType,
    pub entry_id: String,
    pub timestamp: u64,
}

#[derive(Debug, Clone, Copy, PartialEq, Eq, Serialize, Deserialize)]
#[serde(rename_all = "snake_case")]
pub enum MemoryEventType {
    Created,
    Updated,
    Deleted,
}
